### PR TITLE
Improve accessibility of search form

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

Accessibility testing revealed that the search button is difficult for screen reader users to locate, and that the `title` attribute for the keyword search input is not useful.

#### Relevant ticket(s):

* [GDT-318](https://mitlibraries.atlassian.net/browse/GDT-318)

#### How this addresses that need:

This changes the keyword `title` to 'keyword anywhere' and moves the search button to the bottom of the form, after all of the advanced search dropdowns. It also updates some corresponding CSS, including removing the `border-radius` override for the `btn` class, so the search button looks more like a button.

#### Side effects of this change:

Fixed margin/padding for results summary panel and mobile filter toggle. (This is unrelated to this change, but I noticed while testing out the changes that the spacing looked odd.)

##### Accessibility

- [x] ANDI or WAVE has been run in accordance to [our guide](https://mitlibraries.github.io/guides/basics/a11y.html).
- [ ] This PR contains no changes to the view layer.
- [ ] New issues flagged by ANDI or WAVE have been resolved.
- [ ] New issues flagged by ANDI or WAVE have been ticketed (link in the Pull Request details above).
- [x] No new accessibility issues have been flagged.

##### New ENV

- [ ] All new ENV is documented in README.
- [ ] All new ENV has been added to Heroku Pipeline, Staging and Prod.
- [x] ENV has not changed.

##### Approval beyond code review

- [x] UXWS/stakeholder approval has been confirmed.
- [ ] UXWS/stakeholder review will be completed retroactively.
- [ ] UXWS/stakeholder review is not needed.

##### Additional context needed to review

UXWS will do this as part of QA, but it may be worth checking that everything looks okay in a smaller viewport.

##### Code

- [ ] I have confirmed that the code works as intended.
- [ ] Any CodeClimate issues have been fixed or confirmed as
added technical debt.

##### Documentation

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message).
- [ ] The documentation has been updated or is unnecessary.
- [ ] New dependencies are appropriate or there were no changes.

##### Testing

- [ ] There are appropriate tests covering any new functionality.
- [ ] No additional test coverage is required.

[GDT-318]: https://mitlibraries.atlassian.net/browse/GDT-318?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/views/search/_form.html.erb [hunk 1]
```diff
@@ -34,12 +28,9 @@ end
 <form id="basic-search" class="form-horizontal basic-search" action="<%= results_path %>" method="get" role="search">
   <div class="form-group">
     <input id="basic-search-main" type="search"
-           class="field field-text basic-search-input <%= "required" if search_required %>" name="q" title="<%= label %>"
-           placeholder="Enter your search"
+           class="field field-text basic-search-input <%= "required" if search_required %>" name="q"
+           title="Keyword anywhere" placeholder="Enter your search"
```
Comment (JPrevost) [inline app/views/search/_form.html.erb:32]: I hadn't noticed previously that when we expand to show additional form fields we change the placeholder text for this field. I understand why we do that, but it feels a bit odd and we may want to revisit (later) how to handle this (including possibly just throwing "advanced search" into it's own view which I don't love but does make things simpler for a few edge cases).

File: app/views/search/_form.html.erb [hunk 2]
```diff
@@ -220,6 +211,10 @@ end
       </div>
     </details>
   </div>
+
+  <div class="basic-search-submit">
+    <button type="submit" class="btn button-primary">Search</button>
```
Comment (JPrevost) [inline app/views/search/_form.html.erb:216]: I just checked our style guide for other button color options and none of them change the non-hover state. I swear Bento used to have green button links, but maybe they were local overrides. Oh well. One day we'll have a non-greyscale UI element to use.

**Review by JPrevost — APPROVED** [review]

File: app/views/search/_form.html.erb (hunk 1 above)
Comment (jazairi) [reply app/views/search/_form.html.erb:32]: We used to do it for all of the dropdowns as well. I think it's bad practice (usability testing confirmed that suspicion), and I'm taking this opportunity to remove the last instance of it.

File: app/views/search/_form.html.erb (hunk 2 above)
Comment (jazairi) [reply app/views/search/_form.html.erb:216]: Yeah, I saw the same. We could implement a custom style here if it's just not clear that it's a button, but tbh I think eliminating the `border-radius: 0` override will go a long way toward making it more button-like.

Comment (JPrevost) [reply app/views/search/_form.html.erb:216]: Yeah I think it's pretty clear it's a button and what it does, I also just feel like we convinced ourselves years ago that all colors are bad somehow.

**Comment by jazairi** [comment]
FYI, I requested review from Darcy on the PR app, so I'm waiting to merge until after I hear from her.

File: app/views/search/_form.html.erb (hunk 1 above)
Comment (JPrevost) [reply app/views/search/_form.html.erb:32]: Can you clarify what you commented here? I still see this behavior on this PR for all of the various expandable form components. Are you referring to something other than the placeholder for keyword searching changing when expanding/collapsing?

Comment (jazairi) [reply app/views/search/_form.html.erb:32]: Oh, I was actually talking about labels, not placeholder text. We used to change the text of the dropdown labels when they were open (e.g., "Geospatial bounding box search" became "Close geospatial bounding box search"). The keyword input label was the last instance of that, [changed here](https://github.com/MITLibraries/timdex-ui/pull/188/files#diff-9c34bd582e8a3cd8992d37b0395c42f94dec67ce833669687f6f847956719eedL2-L7).

I don't care as much about placeholder text, but I'd be perfectly happy to leave it as 'Keyword anywhere' in all form states. I'm not sure 'Enter your search' is adding any value here.

Sorry for the confusion; I only realized after your question how confusing my comment was. 🙂

Comment (JPrevost) [reply app/views/search/_form.html.erb:32]: Gotcha, I see what you were referring to and yes that has been a good change.

I don't feel strongly enough about the placeholder text to override any previous decisions on it. I find it unexpected, but it isn't inherently wrong.

Comment (jazairi) [reply app/views/search/_form.html.erb:32]: Not inherently wrong, though, but I get the sense that 'Keyword anywhere' is more useful regardless of form state. (But, like you, probably not enough to make a change here.)

**Comment by jazairi** [comment]
UXWS has approved this change.

**Pull request merged**